Apps/VisionOS/VisionOS/CommonViews/TutorialsMenuView.swift

```diff
@@ -0,0 +1,45 @@
+import SwiftUI
+
+@ViewBuilder func button(
+    forLink link: InlineNavigationLink,
+    state: AppState
+) -> some View {
+    Button {
+        state.navigationPath = [link]
+    } label: {
+        Text(link.menuTitle)
+        if state.visitedLinks.contains(link) {
+            Image(systemName: "checkmark.circle.fill")
+        }
+    }
+}
+
+func getEnabledNavigationLinks(_ state: AppState) -> [InlineNavigationLink] {
+    if state.workspaceSettings.isSet(), state.profile.loggedIn {
+        return InlineNavigationLink.allCases
+    }
+
+    if state.workspaceSettings.isSet() {
+        return InlineNavigationLink.preIdentifyLinks
+    }
+
+    return InlineNavigationLink.preInitializationLinks
+}
```

Comment:
This code seems un-related to `TutorialsMenuView`. If that's true, could we move this code into another file?

Reply:
It belongs here. The button view builder decides whether to show a checkmark in the menu entry based on whether the said link is visited. The `getEnabledNavigationLinks` we can argue whether it belongs here or in the shared view model. Since it is only used here, it seems appropriate to keep it here.